// react-instantsearch-hooks-web/dist/es/widgets/SearchBox.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["queryHook"];
import React, { useEffect, useRef, useState } from 'react';
import { useSearchBox } from 'react-instantsearch-hooks';
import { SearchBox as SearchBoxUiComponent } from "../ui/SearchBox.js";
export function SearchBox(_ref) {
  var queryHook = _ref.queryHook,
      props = _objectWithoutProperties(_ref, _excluded);

  var _useSearchBox = useSearchBox({
    queryHook: queryHook
  }, {
    $$widgetType: 'ais.searchBox'
  }),
      query = _useSearchBox.query,
      refine = _useSearchBox.refine,
      isSearchStalled = _useSearchBox.isSearchStalled;

  var _useState = useState(query),
      _useState2 = _slicedToArray(_useState, 2),
      value = _useState2[0],
      setValue = _useState2[1];

  var inputRef = useRef(null);

  function onReset() {
    setValue('');
  }

  function onChange(event) {
    setValue(event.currentTarget.value);
  } // Track when the value coming from the React state changes to synchronize
  // it with InstantSearch.


  useEffect(function () {
    if (query !== value) {
      refine(value);
    } // We don't want to track when the InstantSearch query changes.
    // eslint-disable-next-line react-hooks/exhaustive-deps

  }, [value, refine]); // Track when the InstantSearch query changes to synchronize it with
  // the React state.

  useEffect(function () {
    // We bypass the state update if the input is focused to avoid concurrent
    // updates when typing.
    if (document.activeElement !== inputRef.current && query !== value) {
      setValue(query);
    } // We don't want to track when the React state value changes.
    // eslint-disable-next-line react-hooks/exhaustive-deps

  }, [query]);
  var uiProps = {
    inputRef: inputRef,
    isSearchStalled: isSearchStalled,
    onChange: onChange,
    onReset: onReset,
    value: value,
    translations: {
      submitTitle: 'Submit the search query.',
      resetTitle: 'Clear the search query.'
    }
  };
  return /*#__PURE__*/React.createElement(SearchBoxUiComponent, _extends({}, props, uiProps));
}
